src/aiida/engine/daemon/execmanager.py

```diff
@@ -440,11 +440,19 @@ async def stash_calculation(calculation: CalcJobNode, transport: Transport) -> N
 
     logger_extra = get_dblogger_extra(calculation)
 
+    if calculation.process_type == 'aiida.calculations:core.stash':
+        remote_node = load_node(calculation.inputs.source_node.pk)
+        uuid = remote_node.uuid
+        source_basepath = Path(remote_node.get_remote_path())
+    else:
+        uuid = calculation.uuid
+        source_basepath = Path(calculation.get_remote_workdir())
+
     stash_options = calculation.get_option('stash')
     stash_mode = stash_options.get('stash_mode')
     source_list = stash_options.get('source_list', [])
-    uuid = calculation.uuid
-    source_basepath = Path(calculation.get_remote_workdir())
+    target_base = Path(stash_options['target_base'])
+    dereference = stash_options.get('dereference', False)
```

Comment:
It is not nice that we need to know here (or anywhere where we use the options) how the stash options are validated in terms of types to access and use the dictionary correctly. It would be nicer if we could somehow use a typed dict to impose structure.

```python
from typing import TypedDict
class StashOption(TypedDict):
    target_base: Union[str, Path] 
    source_list: Union[list[str, Path], tuple[str, Path]]
    stash_mode: str = StashMode.COPY.value
    dereference: bool = False
```
I think this is a general issue with the options and I am also not sure if TypedDict works here so I would not block this PR because of this but I wanted to mention it.

Reply:
fair enough